tests/unit/test_field_generator.py

```diff
@@ -0,0 +1,152 @@
+    instantiates the default FieldDataGenerator plugins and pre-populates the
+    dict from which the get_field_data_generator method retrieves them. It
+    uses the get_field_instance_dict_from_pydantic_class method to pre-load
+    some valid field data that the FieldDataGenerator subclasses can return
+    via the get_field_data method.
+    """
+    def __init__(self):
+        field_data = {
+            "bugzilla_field_data": [
+                get_field_instance_data_dict_from_field_class(BugzillaField)
+            ],
+            "jira_field_data": [
+                get_field_instance_data_dict_from_field_class(JiraField)
+            ]
+        }
+
+        bzfg = BugzillaFieldDataGenerator({})
+        jfg = JiraFieldDataGenerator({})
+        for instance in bzfg, jfg:
+            instance.field_data = field_data
+
+        self.field_data_generators = {
+            BUGZILLA: bzfg,
+            JIRA: jfg
+        }
+
+
+@pytest.fixture(autouse=True)
+def patch_data_generator_factory():
+    """This fixture replaces the field_data_factory used in the field_generator
+    module. In its place we put an instance of StubbedFieldGeneratorFactory.
+    """
+    patcher = patch("bugjira.field_generator.field_data_factory",
+                    StubbedFieldGeneratorFactory())
+    patcher.start()
+    yield
+    patcher.stop()
+
+
+@pytest.mark.parametrize("generator_type,field_class,data_generator_class",
+                         [
+                             (BUGZILLA, BugzillaField,
+                              BugzillaFieldDataGenerator),
+                             (JIRA, JiraField,
+                              JiraFieldDataGenerator)
+                         ])
+def test_field_generator_init(generator_type, field_class,
+                              data_generator_class):
+    """
+    GIVEN the FieldGenerator class
+    WHEN we initialize it
+    THEN the instance's config, field_data_generator, and field_class
+        attributes should be set to expected values
+    """
+    config = {"expected": "config"}
+    fg = FieldGenerator(generator_type, config)
+    assert fg.config == config
+    assert isinstance(fg.field_data_generator, data_generator_class)
+    assert fg.field_class == field_class
+
+
+@pytest.mark.parametrize("generator_type,expected_value",
+                         [(BUGZILLA, BugzillaField),
+                          (JIRA, JiraField)])
+def test_get_field_class(generator_type, expected_value):
+    """
+    GIVEN an instance of the FieldGenerator class
+    WHEN we call _get_field_class with a valid generator_type
+    THEN the correct BugjiraField subclass associated with the type is returned
+    """
+    fg = FieldGenerator(generator_type, {})
+    assert fg._get_field_class(generator_type) == expected_value
+
+
+def test_get_field_class_bad_key():
+    """
+    GIVEN an instance of the FieldGenerator class
+    WHEN we call _get_field_class with an invalid generator_type
```

Comment:
(non-clocking) question: Why this generator type is invalid? I've double checked with test_get_field_class generator_type BUGZILLA is the same as the one at L117

Reply:
BUGZILLA is only used to instantiate FieldGenerator here. "bad key" is the invalid generator_type parameter to the _get_field_class method. I'll add a comment to make this clearer in the test.